File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
const solution = (s) => {
  const result = [];
  const lastIndex = {};

  for (let i = 0; i < s.length; i++) {
    const char = s[i];
    if (lastIndex[char] !== undefined) {
      result.push(i - lastIndex[char]);
    } else {
      result.push(-1);
    }
    lastIndex[char] = i;
  }

  return result;
};

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
class Solution {
    public int[] solution(String s) {
        int n = s.length();
        int[] answer = new int [n];
        int[] lastIndex = new int[26];
        for (int i = 0; i< 26; i++){
            lastIndex[i] = -1;
        }
        for(int i = 0; i < n; i++){
            char c = s.charAt(i);
            
            int index = c - 'a';
            
            if(lastIndex[index] == -1){
                answer[i] = -1;
            }else {
                answer[i] = i - lastIndex[index];
                
            }
            lastIndex[index] = i;
        }
        return answer;
    }
}